src/System.Private.Xml/tests/Xslt/XslTransformApi/CXslTransform.cs

```diff
@@ -384,31 +384,32 @@ public CXmlResolverTest(ITestOutputHelper output) : base(output)
         [Theory]
         public void XmlResolver1(InputType inputType, ReaderType readerType, TransformType transformType, DocType docType)
         {
-            AppContext.SetSwitch("Switch.System.Xml.AllowDefaultResolver", true);
-
-            string expected = @"<?xml version=""1.0"" encoding=""utf-8""?><result><fruit>Apple</fruit><fruit>orange</fruit></result>";
-
-            try
+            using (new AllowDefaultResolverContext())
             {
-                if (LoadXSL("XmlResolver_Main.xsl", inputType, readerType) == 1)
+                string expected = @"<?xml version=""1.0"" encoding=""utf-8""?><result><fruit>Apple</fruit><fruit>orange</fruit></result>";
+
+                try
                 {
-                    xslt.XmlResolver = null;
-                    if (Transform("fruits.xml", transformType, docType) == 1)
+                    if (LoadXSL("XmlResolver_Main.xsl", inputType, readerType) == 1)
                     {
-                        VerifyResult(expected);
-                        return;
+                        xslt.XmlResolver = null;
+                        if (Transform("fruits.xml", transformType, docType) == 1)
+                        {
+                            VerifyResult(expected);
+                            return;
+                        }
+                        else
+                            Assert.True(false);
                     }
-                    else
-                        Assert.True(false);
                 }
-            }
-            catch (Exception e)
-            {
-                _output.WriteLine("Should not throw error loading stylesheet with include/import when resolver property is set to NULL!");
-                _output.WriteLine(e.ToString());
+                catch (Exception e)
+                {
+                    _output.WriteLine("Should not throw error loading stylesheet with include/import when resolver property is set to NULL!");
+                    _output.WriteLine(e.ToString());
+                    Assert.True(false);
+                }
```

Comment:
Rather than catching and asserting, seems like it should just let the exception propagate out to fail the test.  If for some reason we need to include more information, the exception could be caught and wrapped in another exception that includes the first and more details.